extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/discovery/AstroDiscoveryService.java

```diff
@@ -31,11 +38,14 @@
  * @author Gerhard Riegler - Initial Contribution
  * @author Stefan Triller - Use configured location
  */
+@Component(service = DiscoveryService.class, immediate = true, configurationPid = "discovery.astro")
 public class AstroDiscoveryService extends AbstractDiscoveryService {
     private final Logger logger = LoggerFactory.getLogger(AstroDiscoveryService.class);
     private static final int DISCOVER_TIMEOUT_SECONDS = 30;
     private static final int LOCATION_CHANGED_CHECK_INTERVAL = 60;
     private LocationProvider locationProvider;
+    private Locale locale;
+    private DiscoveryResultI18nUtil discoveryResultI18nUtil;
```

Comment:
These members seem to be common for all i18n discovery services. Couldn't you move them to a new class "AbstractI18nDiscoveryService" or something like that from which the actual implementation inherits? This would reduce code duplication.

Reply:
That is probably doable. Will study that.

Reply:
I created a new abstract class that replaced the util class.